### PR TITLE
Refactor v1

### PR DESCRIPTION
**Pull request by matgar26**

### CODE REVIEW

File: Sources/FeedManager.swift [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        
+        public init(feedId: String, options: FeedClientOptions = FeedClientOptions(archived: .exclude)) throws {
+            Task {
+                self.feedModule = try await FeedModule(feedId: feedId, options: options)
+                registerForAppLifecycleNotifications()
+            }
+        }
+        
+        deinit {
+            unregisterFromAppLifecycleNotifications()
```
Comment (cjbell) [inline Sources/FeedManager.swift]: Nit: I think I'd name this method deregister vs unregister

File: README.md [hunk 2]
```diff
@@ -119,47 +92,20 @@ You can now start using the SDK:
 ``` swift
 import Knock
 
-knockClient = try! Knock(publishableKey: "your-pk", userId: "user-id")
+// Setup the shared Knock instance as soon as you can. 
+try? Knock.shared.setup(publishableKey: "your-pk", pushChannelId: "user-id")
```
Comment (cjbell) [inline README.md]:
```suggestion
try? Knock.shared.setup(publishableKey: "your-pk", pushChannelId: "apns-push-channel-id")
```

File: README.md [hunk 3]
```diff
@@ -119,47 +92,20 @@ You can now start using the SDK:
 ``` swift
 import Knock
 
-knockClient = try! Knock(publishableKey: "your-pk", userId: "user-id")
+// Setup the shared Knock instance as soon as you can.
```
Comment (cjbell) [inline README.md]: We might want to add a note that the `pushChannelId` is optional?

File: Sources/Knock.swift [hunk 4]
```diff
@@ -6,38 +6,72 @@
...
+    internal lazy var messageModule = MessageModule()
+    internal lazy var channelModule = ChannelModule()
+    internal lazy var logger = KnockLogger()
+    
+    /**
+    Sets up the shared Knock instance. Make sure to call this as soon as you can. Preferrably in your AppDelegate.
+
+     - Parameters:
+        - publishableKey: Your public API key
+        - pushChannelId:
```
Comment (cjbell) [inline Sources/Knock.swift]: 👀 missing doc for this

**Review by cjbell — APPROVED** [review]
few small comments, not blocking 💯

**Comment by matgar26** [comment]
@cjbell thanks for the feedback. I've implemented your suggestions in the latest commit

**Review by cjbell — APPROVED** [review]

**Pull request merged**